distributed-counter/src/main/java/org/infinispan/counter/Counter.java

```diff
@@ -0,0 +1,84 @@
+package org.infinispan.counter;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An interface to interact to a counter.
+ * <p>
+ * It allows to update the counter by adding and removing. All the operations are perform asynchronously and they return
+ * a {@link CompletableFuture} when completed.
+ * <p>
+ * The implementation may support weakly consistent value via {@link #weakGetValue()}.
+ *
+ * @author Pedro Ruivo
+ * @since 9.0
+ */
+public interface Counter {
```

Comment:
`AtomicLong` is class, this is iface. I see no gain in mixing those, and it's better when users are clearly aware that a counter is a distributed one rather than super-fast atomics.